backend/amp-live-list.go

```diff
@@ -119,20 +149,38 @@ func readStatus(r *http.Request) int {
 	return result
 }
 
-func createPage(newStatus int, timestamp time.Time) Page {
+func createMetadata(r *http.Request) []BlogPosting {
+	result := make([]BlogPosting, 0)
+	for i := 0; i < len(blogs); i++ {
+		result = append(result, BlogPosting{"BlogPosting",
+			blogs[i].Heading,
+			buildSourceOrigin(r.Host) + "/samples_templates/live_blog/#" + blogs[i].ID,
+			blogs[i].MetadataTimestamp,
+			ArticleBody{"Text"},
+			Publisher{"Organization", "AMPByExample",
+				Image{"ImageObject", "https://ampbyexample.com/img/favicon.png", "512", "512"}},
+			Image{"ImageObject", blogs[i].Image, "853", "1280"},
+		})
+	}
+	return result
+}
+
+func createPage(newStatus int, timestamp time.Time, r *http.Request) Page {
 	if newStatus > len(blogs) {
 		newStatus = len(blogs)
 	}
 	blogItems := getBlogEntries(newStatus, timestamp)
 	score := createScore(newStatus, 0)
-	return Page{BlogItems: blogItems, FootballScore: score}
+	metadata, _ := json.Marshal(createMetadata(r))
+	return Page{BlogItems: blogItems, FootballScore: score, BlogMetadata: string(metadata)}
 }
 
 func renderSample(w http.ResponseWriter, r *http.Request, filePath string) {
-	t, _ := template.ParseFiles(filePath)
+	t, error := template.ParseFiles(filePath)
+	log.Println(error)
```

Comment:
correctly handle the error, `panic(error)` is OK if you parse the template on startup -> don't parse the files with every request, instead cache the parsed template.